### DISCUSSION TITLE
Can we do upsert using DRF serializers

### DISCUSSION THREAD

**Discussion by lokeshwarobuli** [encode/django-rest-framework#8888] [General]
Is there any way to override the serializers to do upsert similar to Django's update_or_create method ? for a list of dictionaries

**Comment by auvipy** [comment]
shouldn't we try to do that on orm level?

**Reply by lokeshwarobuli** [reply]
can you give an example, on how to do that at an ORM level in Django?
I have tried couple of things but no luck

**Reply by lokeshwarobuli** [reply]
@auvipy any luck for a code sample ?

**Reply by nachollica** [reply]
i don't know your particular use case, so i will try to give a generic answer.

in first place, remember that Serializers are meant to validate the data, and "translate" data (for example, from a row into your DB to a JSON for a response). the task of actually creating/updating a resource in the DB is still the responsibility of the ORM.

you're trying to create or update rows. so i assume two things: you're working with a ModelSerializer, and a view that receives a POST/PUT/PATCH request. something you can do here is to override the `create ` and/or `update` methods of your Serializer to call the method you mentioned `YourModel.objects.update_or_create`.

In relation with the "for a list of dictionaries" part of your question, remember the `many=True` argument of the Serializers.

Refer to the docs for more details and examples: https://www.django-rest-framework.org/api-guide/serializers/#saving-instances

**Reply by lokeshwarobuli** [reply]
@nchz Thanks for the clarification !!
